### PR TITLE
MSEARCH-72: Throw more explicit error if tenant is not initialized yet

### PR DESCRIPTION
**Pull request by bohdan-suprun**
When kafka listener starts consuming events from the topic but tenant is not initialized yet we throw a `TenantNotInitializedException` vs a sql exception which confuses folks.

### CODE REVIEW

File: src/main/java/org/folio/search/integration/error/KafkaErrorHandler.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.springframework.messaging.Message;
+import org.springframework.stereotype.Component;
+
+@Component
+public class KafkaErrorHandler implements KafkaListenerErrorHandler {
+  @Override
+  public Object handleError(Message<?> message, ListenerExecutionFailedException exception) {
+    if (isTenantNotInitializedYet(exception)) {
+      throw new TenantNotInitializedException(getTenantNames(message));
+    } else {
```
Comment (pfilippovEPAM) [inline src/main/java/org/folio/search/integration/error/KafkaErrorHandler.java:21]: else condition can be removed

**Review by pfilippovEPAM — APPROVED** [review]

File: src/main/java/org/folio/search/integration/error/KafkaErrorHandler.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    // which is treated as a grammar exception.
+    // It is possible to have false positives here, when an invalid SQL is used
+    // but anyway it results in retrying the consume operation
+    return getThrowableList(exception).stream().anyMatch(SQLGrammarException.class::isInstance);
+  }
+
+  @SuppressWarnings("unchecked")
+  private String[] getTenantNames(Message<?> message) {
+    if (!isSupportedType(message)) {
+      return new String[0];
```
Comment (hjiebsco) [inline src/main/java/org/folio/search/integration/error/KafkaErrorHandler.java:38]: Curious, in this case, Is it still a `TenantNotInitializedException` even though there are no tenant names?

File: src/main/java/org/folio/search/integration/error/KafkaErrorHandler.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      throw exception;
+    }
+  }
+
+  private boolean isTenantNotInitializedYet(ListenerExecutionFailedException exception) {
+    // In case when the schema for tenant is not created
+    // DB will throw an error saying that a table does not exists
+    // which is treated as a grammar exception.
+    // It is possible to have false positives here, when an invalid SQL is used
+    // but anyway it results in retrying the consume operation
```
Comment (hjiebsco) [inline src/main/java/org/folio/search/integration/error/KafkaErrorHandler.java:31]: Would those potential false positives make the logs more confusing?

**Review by hjiebsco — APPROVED** [review]

File: src/main/java/org/folio/search/integration/error/KafkaErrorHandler.java (hunk 2 above)
Comment (bohdan-suprun) [reply src/main/java/org/folio/search/integration/error/KafkaErrorHandler.java]: Make sense, removed this part.

File: src/main/java/org/folio/search/integration/error/KafkaErrorHandler.java (hunk 3 above)
Comment (bohdan-suprun) [reply src/main/java/org/folio/search/integration/error/KafkaErrorHandler.java:31]: I'm now specifying the original cause, so we can debug it when needed.

File: src/main/java/org/folio/search/integration/error/KafkaErrorHandler.java (hunk 1 above)
Comment (bohdan-suprun) [reply src/main/java/org/folio/search/integration/error/KafkaErrorHandler.java]: changed

**Pull request merged**